### ISSUE TITLE
How to write definition file for react component

### ISSUE THREAD

**Issue by thanzen** [TypeStrong/atom-typescript#442]
I tried to use react-quill in my component. The code is as below:
```ts
///<reference path="../../libs/typings/react.d.ts" />
///<reference path="../../libs/typings/react-quill.d.ts" />
import React = require('react');
import ReactQuill = require('react-quill');

interface Props { }
interface State { }

class Editor extends React.Component<Props, State>{
    getInitialState() {
        return { value };
    }
    render() {
        return (
            <ReactQuill theme="snow" />
            );
    }
}
```
And the definition I made is  :
```ts
/// <reference path="react-jsx.d.ts"/>
declare var ReactQuill: JSX.ElementClass;
declare module 'react-quill'{
    export = ReactQuill
}
```

The atom-typescript did not seem to compile the file, though there was no errors. 
 
![image](https://cloud.githubusercontent.com/assets/1901357/8514185/254bee4a-233d-11e5-8f76-030f811b95d4.png)

**Comment by basarat** [comment]
> The atom-typescript did not seem to compile the file, though there was no errors.

I don't see a `tsconfig.json` in that screenshot. Can you share that.

TIP : Just to get you on your way : try restarting atom

**Comment by basarat** [comment]
Just did a quick test and it worked fine (in that it did do an *emit*): 

![image](https://cloud.githubusercontent.com/assets/874898/8517031/22013066-23ff-11e5-8cba-80fb57ed719f.png)

**Comment by thanzen** [comment]
Right I see. But it did not output this line:
import ReactQuill = require('react-quill');
Btw, this is my tsconfig:
{
    "version": "1.5.0-beta",
    "compilerOptions": {
        "target": "es5",
        "module": "commonjs",
        "isolatedModules": false,
        "jsx": "react",
        "experimentalDecorators": true,
        "emitDecoratorMetadata": true,
        "declaration": false,
        "noImplicitAny": false,
        "removeComments": true,
        "noLib": false,
        "preserveConstEnums": true,
        "suppressImplicitAnyIndexErrors": true
    },
    "filesGlob": [
        "./**/*.ts",
        "./**/*.tsx",
        "!./node_modules/**/*"
    ],
    "files": [
        "./libs/typings/globals.ts",
        "./libs/typings/react-jsx.d.ts",
        "./libs/typings/react-quill.d.ts",
        "./libs/typings/react.d.ts",
        "./src/components/Editor.tsx"
    ]
}

**Comment by basarat** [comment]
> But it did not output this line:

You've found a bug in Microsoft/TypeScript. Reported https://github.com/Microsoft/TypeScript/issues/3759 :rose:

**Comment by basarat** [comment]
Fixed with `5.0.16`

**Comment by thanzen** [comment]
Super!
Thanks.
